x/gov/types/expected_keepers.go

```diff
@@ -48,3 +48,16 @@ type BankKeeper interface {
 	SendCoinsFromAccountToModule(ctx sdk.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins) error
 	BurnCoins(ctx sdk.Context, name string, amt sdk.Coins) error
 }
+
+// Event Hooks
+// These can be utilized to communicate between a governance keeper and another
+// keepers.
+
+// GovHooks event hooks for governance proposal object (noalias)
+type GovHooks interface {
+	AfterProposalSubmission(ctx sdk.Context, proposalID uint64)                            // Must be called after proposal is submitted
+	AfterProposalDeposit(ctx sdk.Context, proposalID uint64, depositorAddr sdk.AccAddress) // Must be called after a deposit is made
+	AfterProposalVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress)        // Must be called after a vote on a proposal is cast
+	AfterProposalInactive(ctx sdk.Context, proposalID uint64)                              // Must be called when proposal become inactive
+	AfterProposalActive(ctx sdk.Context, proposalID uint64)                                // Must be called when proposal become active
```

Comment:
I would propose to put the struct as 2nd argument everywhere:
```go
AfterProposalVote(ctx sdk.Context, vote Vote)
AfterProposalActive(ctx sdk.Context, proposal Proposal)
```

etc, so that users can access all fields if needed.

Reply:
Currently, the hooks in the staking module don't pass around the full structs, and only pass the minimal information needed such that the recipient can query the staking module for more information if desired.

For example, in the new delegation hook, it just passes the validator address and delegator address.  The recipient can then use `stakingKeeper.GetDelegation(valAddr, delAddr)` to get the delegation struct.

I was trying to keep the same standard for these as well, but I'm not sure the pros/cons of doing this vs the entire struct (in which case, we should switch it for staking hooks as well imo).